src/sentry/api/endpoints/relocation.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import re
+
+from django.db import router
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, region_silo_endpoint
+from sentry.api.permissions import SuperuserPermission
+from sentry.models.files.file import File
+from sentry.models.relocation import Relocation, RelocationFile
+from sentry.models.user import MAX_USERNAME_LENGTH
+from sentry.services.hybrid_cloud.user.service import user_service
+from sentry.utils.db import atomic_transaction
+
+# Relocation input files are uploaded as tarballs, and chunked and stored using the normal
+# `File`/`AbstractFile` mechanism, which has a hard limit of 2GiB, because we need to represent the
+# offset into it as a 32-bit int. This means that the largest tarball we are able to import at this
+# time is 2GiB. When validating this tarball, we will need to make a "composite object" from the
+# uploaded blobs in Google Cloud Storage, which has a limit of 32 components. Thus, we get our blob
+# size of the maximum overall file size (2GiB) divided by the maximum number of blobs (32): 65536MiB
+# per blob.
+RELOCATION_BLOB_SIZE = int((2**31) / 32)
+
+ERR_DUPLICATE_RELOCATION = "An in-progress relocation already exists for this owner"
+ERR_FEATURE_DISABLED = "This feature is not yet enabled"
+_filename_re = re.compile(r"[\n\t\r\f\v\\]")
+
+logger = logging.getLogger(__name__)
+
+
+class RelocationPostSerializer(serializers.Serializer):
+    file = serializers.FileField(required=True)
+    orgs = serializers.ListField(required=True, allow_empty=False)
+    owner = serializers.CharField(
+        max_length=MAX_USERNAME_LENGTH, required=True, allow_blank=False, allow_null=False
+    )
+
+
+class RelocationMixin:
+    @staticmethod
+    def post_relocation(request, logger):
+        if not features.has("relocation:enabled"):
+            return Response({"detail": ERR_DUPLICATE_RELOCATION}, status=501)
```

Comment:
```suggestion
            return Response({"detail": ERR_DUPLICATE_RELOCATION}, status=400)
```

We shouldn't be returning 500s to user requests unless we've fallen down.

Reply:
Changed to a 400.